File: mobile/android/app/src/main/java/uz/rento/domain/model/Message.kt
package uz.rento.domain.model

/**
 * Message — xabar domain modeli
 */
data class Message(
    val id: String,
    val roomId: String,
    val senderId: String,
    val content: String?,
    val messageType: MessageType,
    val mediaUrl: String?,
    val metadata: Map<String, Any>?,
    val isRead: Boolean,
    val readAt: String?,
    val createdAt: String
) {
    /** Foydalanuvchi yuborgan xabarmi */
    fun isMine(currentUserId: String): Boolean = senderId == currentUserId

    /** Rasm xabari */
    val isImage: Boolean get() = messageType == MessageType.IMAGE

    /** Lokatsiya xabari */
    val isLocation: Boolean get() = messageType == MessageType.LOCATION
}

/**
 * MessageType — xabar turi
 */
enum class MessageType(val value: String) {
    TEXT("text"),
    IMAGE("image"),
    LOCATION("location"),
    CONTACT("contact");

    companion object {
        fun fromValue(value: String): MessageType {
            return entries.firstOrNull { it.value == value } ?: TEXT
        }
    }
}

File: services/rento-api/src/Rento.Core/Entities/Message.cs
using System.ComponentModel.DataAnnotations.Schema;
using Rento.Core.Common;

namespace Rento.Core.Entities;

[Table("messages")]
public class Message
{
    public Guid Id { get; set; }
    public Guid RoomId { get; set; }
    public Guid SenderId { get; set; }
    public string? Content { get; set; }
    public string MessageType { get; set; } = Rento.Core.Common.MessageType.Text;
    public string? MediaUrl { get; set; }
    public string? Metadata { get; set; }
    public bool IsRead { get; set; }
    public DateTime? ReadAt { get; set; }
    public DateTime CreatedAt { get; set; }

    public ChatRoom Room { get; set; } = null!;
    public User Sender { get; set; } = null!;
}
